obspy/core/tests/test_waveform_plugins.py

```diff
@@ -141,6 +146,8 @@ def test_readAndWrite(self):
                         self.assertEqual(st[0].id, ".MANZ1..EHE")
                     elif format not in ['WAV']:
                         self.assertEqual(st[0].id, "BW.MANZ1.00.EHE")
+                    if format not in WAVEFORM_ACCEPT_BYTEORDER:
```

Comment:
Assuming we want to merge this: Instead of breaking the loop I would prefer it if the byteorder list that is being looped over would be dependent on the `WAVEFORM_ACCEPT_BYTEORDER` flag. That would be easier to reason about and the test is hard enough to understand in any case, e.g.

```python
for native_byteorder in ['<', '>']:
    for byteorder in ['<', '>', '='] if format in WAVEFORM_ACCEPT_BYTEORDER else [None]:
        ...
```